packages/subgraph/src/packages/addresslist/addresslist-voting.ts

```diff
@@ -111,39 +111,43 @@ export function handleVoteCast(event: VoteCast): void {
     let contract = Addresslist.bind(event.address);
     let vote = contract.try_getProposal(event.params.proposalId);
     if (!vote.reverted) {
-      let voteCount = vote.value.value8.plus(
-        vote.value.value9.plus(vote.value.value10)
-      );
       let yes = vote.value.value8;
+      let no = vote.value.value9;
+      let abstain = vote.value.value10;
+      let voteCount = yes.plus(no.plus(abstain));
+
       proposalEntity.yes = yes;
-      proposalEntity.no = vote.value.value9;
-      proposalEntity.abstain = vote.value.value10;
+      proposalEntity.no = no;
+      proposalEntity.abstain = abstain;
       proposalEntity.voteCount = voteCount;
 
-      // check if the current vote results meet
-      // the conditions for the proposal to pass:
-      // - total support    :  N_yes / N_total >= total support threshold
-      // - relative support :  N_yes / (N_yes + N_no) >= relative support threshold
+      // check if the current vote results meet the conditions for the proposal to pass:
+      // - participation      :  (N_yes + N_no + N_abstain) / N_total > minimal participation
+      // - worst case support :  N_yes / (N_total - N_abstain) > support threshold
 
       // expect a number between 0 and 100
-      // where 0.35 => 35
       let currentParticipation = voteCount
         .times(BigInt.fromI32(100))
         .div(proposalEntity.totalVotingPower);
-      // expect a number between 0 and 100
-      // where 0.35 => 35
-      let currentSupport = yes.times(BigInt.fromI32(100)).div(voteCount);
+
+      let worstCaseSupport = yes
+        .times(BigInt.fromI32(100))
+        .div(proposalEntity.totalVotingPower.minus(abstain));
+
+      /*
+      let currentSupport = new BigInt(0);
+      if (voteCount.gt(new BigInt(0))) {
+        currentSupport = yes.times(BigInt.fromI32(100)).div(yes.plus(no));
+      }
+      */
```

Comment:
do we need this commented code here?

Reply:
Removed in 9828813cf9cc312e6884b420ca65fabe40b4a75a